hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/subprocess/ProcessUtils.java

```diff
@@ -159,20 +162,96 @@ public static ProcessHandle startSubProcessNodeFrom(
-            return builder.start().toHandle();
+            final var handle = builder.start().toHandle();
+            persistPid(metadata, handle);
+            return handle;
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
+    /**
+     * Best-effort shutdown scoped to a node's working directory. This avoids killing nodes
+     * in other networks that share the same numeric node id (e.g., multi-network suites).
+     * Falls back to the legacy nodeId-based kill only for the default shared-network scope.
+     *
+     * @param metadata the metadata of the node to stop
+     */
+    public static void destroySubProcessNodeFor(@NonNull final NodeMetadata metadata) {
+        requireNonNull(metadata);
+        final var pidFile = pidFilePath(metadata);
+        boolean stoppedFromPid = false;
+        if (Files.exists(pidFile)) {
+            try {
+                final var pidText = Files.readString(pidFile, UTF_8).trim();
+                if (!pidText.isEmpty()) {
+                    final var pid = Long.parseLong(pidText);
+                    final var handleOpt = ProcessHandle.of(pid);
+                    if (handleOpt.isPresent()) {
+                        final var handle = handleOpt.get();
+                        if (handle.isAlive()) {
+                            log.info(
+                                    "Destroying node{} with PID '{}' from {} (Alive? {})",
+                                    metadata.nodeId(),
+                                    pid,
+                                    pidFile,
+                                    "Yes");
+                            handle.destroy();
+                            if (handle.isAlive()) {
+                                handle.destroyForcibly();
+                            }
+                            stoppedFromPid = true;
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                log.warn("Failed to stop node{} using pid file {}: {}", metadata.nodeId(), pidFile, e.toString());
+            }
+        }
+        // Retain legacy behavior for the default shared-network scope only.
+        if (!stoppedFromPid && isDefaultScope(metadata)) {
+            destroyAnySubProcessNodeWithId(metadata.nodeId());
+        }
+        try {
+            Files.deleteIfExists(pidFile);
+        } catch (IOException e) {
+            log.warn("Failed to delete pid file {} after shutdown: {}", pidFile, e.toString());
+        }
+    }
+
+    private static boolean isDefaultScope(@NonNull final NodeMetadata metadata) {
+        // Default shared-network working dirs use the "hapi-test" scope folder.
+        final var workingDir = metadata.workingDirOrThrow();
+        final var parent = workingDir.getParent();
+        return parent != null
+                && parent.getFileName() != null
+                && "hapi-test".equals(parent.getFileName().toString());
```

Comment:
I suspect we won't want to determine scope this way in the future, but it works fine for now